content/docs/local/neon-local.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Neon Local
+enableTableOfContents: true
+subtitle: Connect to Neon using Docker containers
+updatedOn: '2025-04-22T22:55:27.453Z'
+---
+
+[Neon Local](https://github.com/neondatabase-labs/neon_local) is a proxy service that creates a local interface to your Neon cloud database. By default, it automatically creates a new database branch when the container starts and deletes it when the container stops.
+
+Your application connects to a local Postgres endpoint, while Neon Local handles routing and authentication to the correct project and branch. This removes the need to update connection strings when working across database branches.
+
+## Docker compose instructions
+
+You can add the Neon Local by adding the following to your `docker-compose.yml` file.
+
+<CodeTabs labels={["Neon serverless driver", "node-postgres"]}>
+
+```shell
+  db:
+    image: neondatabase/neon_local:latest
+    ports:
+      - '5432:5432'
+    environment:
+      NEON_API_KEY: ${NEON_API_KEY}
+      NEON_PROJECT_ID: ${NEON_PROJECT_ID}
+      DRIVER: serverless
+```
+
+```shell
+  db:
+    image: neondatabase/neon_local:latest
+    ports:
+      - '5432:5432'
+    environment:
+      NEON_API_KEY: ${NEON_API_KEY}
+      NEON_PROJECT_ID: ${NEON_PROJECT_ID}
+      DRIVER: postgres
+```
+
+</CodeTabs>
+
+## Docker run instructions
+
+You can also run the Neon Local container with the following Docker run command.
+
+<CodeTabs labels={["Neon serverless driver", "node-postgres"]}>
+
+```shell
+docker run \
+  --name db \
+  -p 5432:5432 \
+  -e NEON_API_KEY= <neon_api_key> \
+  -e NEON_PROJECT_ID=<neon_project_id> \
+  -e DRIVER=serverless \
+  neondatabase/neon_local:latest
+```
+
+```shell
+docker run \
+  --name db \
+  -p 5432:5432 \
+  -e NEON_API_KEY= <neon_api_key> \
+  -e NEON_PROJECT_ID=<neon_project_id> \
+  -e DRIVER=postgres \
+  neondatabase/neon_local:latest
+```
+
+</CodeTabs>
+
+## Connect
+
+Connect using either the Neon [serverless driver](/serverless/serverless-driver) or any other Postgres client.
+
+<CodeTabs labels={["Neon serverless driver", "node-postgres"]}>
+
+```javascript
+import { neon, neonConfig } from '@neondatabase/serverless';
+neonConfig.fetchEndpoint = 'http://db:5432/sql';
+
+const sql = neon('postgres://neon:npg@db:5432/neondb');
```

Comment:
@jeff-at-neon, I've just seen your comment on the blog post. is `no-verify` required wth the serverless driver. In my test app I only needed to add this to the `pg` version. But, if you wanna play it safe and add it to all snippets, that's fine with me.